src/hiddb/asynchronous.py

```diff
@@ -25,337 +42,150 @@ async def create(cls, key, secret):
         await self._machine_login(key, secret)
         return self
 
-
     async def _machine_login(self, key: str, secret: str):
-        url = baseDbUrl
-        path = f"/machine/login"
-        method = "post"
-
         body = {
             "access_key": key,
             "secret_key": secret
         }
-
-        async with aiohttp.ClientSession(url) as session:
-            req = getattr(session, method)
-            async with req(path, json=body, headers=postHeaders) as resp:
-                if resp.status != 200:
-                    raise Exception(f"Status code {resp.status}: {await resp.text()}")
-                self.state.access_token = (await resp.json())['access_token']
-                return self.state.access_token
+        request_data = StdRequest(path=f"/machine/login", method="post", body=body)
+        await self.make_request(request_data)
 
     async def create_database(self, name: str):
-        url = baseDbUrl
-        path = f"/database"
-        method = "post"
-
         body = {
             "database_name": name,
         }
-
-        async with aiohttp.ClientSession(url) as session:
-            req = getattr(session, method)
-            session.headers.update({'Authorization' : f'Bearer {self.state.access_token}'})
-            async with req(path, json=body, headers=postHeaders) as resp:
-                if resp.status != 202 and resp.status != 200:
-                    raise Exception(f"Status code {resp.status}: {await resp.text()}")
-                return await resp.json()
+        request_data = StdRequest(path=f"/database", method="post", body=body)
+        await self.make_request(request_data)
 
     async def list_databases(self):
-        url = baseDbUrl
-        path = f"/database"
-        method = "get"
-
-        async with aiohttp.ClientSession(url) as session:
-            req = getattr(session, method)
-            session.headers.update({'Authorization' : f'Bearer {self.state.access_token}'})
-            async with req(path) as resp:
-                if resp.status != 200:
-                    raise Exception(f"Status code {resp.status}: {await resp.text()}")
-                return await resp.json()
-
-
-    async def get_database(self, id: str):
-        url = baseDbUrl
-        path = f"/database/{id}"
-        method = "get"
-
-        async with aiohttp.ClientSession(url) as session:
-            req = getattr(session, method)
-            session.headers.update({'Authorization' : f'Bearer {self.state.access_token}'})
-            async with req(path) as resp:
-                if resp.status != 200:
-                    raise Exception(f"Status code {resp.status}: {await resp.text()}")
-                return await resp.json()
+        request_data = StdRequest(path=f"/database", method="get")
+        await self.make_request(request_data)
 
+    async def get_database(self, _id: str):
```

Comment:
variables with a _ prefix hint towards the variable being private or should not be used. 
This is not the case here.

Reply:
rename here and following places

Reply:
You'r welcome :)
And yes, I agree, just changed that to avoid the word "id" which is a keyword, but probably "database_id" is the word to put there.